File: DoAnChuyenNganh-main/DoAnCoSo/WebDuLichDaLat/Models/TransportPriceHistory.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebDuLichDaLat.Models
{
    public class TransportPriceHistory
    {
        [Key]
        public int Id { get; set; }

        // Khóa ngoại đến loại phương tiện
        [Required]
        public int TransportOptionId { get; set; }
        public TransportOption TransportOption { get; set; }

        // Khóa ngoại đến địa điểm cũ
        [Required]
        public int LegacyLocationId { get; set; }
        public LegacyLocation LegacyLocation { get; set; }

        // Giá vé áp dụng riêng cho địa điểm đó
        [Column(TypeName = "decimal(18,2)")]
        public decimal Price { get; set; }
    }
}

File: src/main/java/com/example/dalattravel/model/TransportPriceHistory.java
package com.example.DaLattravel.model;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "transport_price_histories", uniqueConstraints = {
    @UniqueConstraint(name = "IX_TransportPrice_LocationTransport", columnNames = {"legacyLocationId", "transportOptionId"})
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TransportPriceHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false)
    private Integer legacyLocationId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "legacyLocationId", insertable = false, updatable = false)
    private LegacyLocation legacyLocation;

    @Column(nullable = false)
    private Integer transportOptionId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "transportOptionId", insertable = false, updatable = false)
    private TransportOption transportOption;

    @Column(precision = 18, scale = 2, nullable = false)
    private BigDecimal price;

    @Builder.Default
    private LocalDateTime updatedAt = LocalDateTime.now();
}
